src/plans/create/CreatePlanWizard.tsx

```diff
@@ -9,44 +10,92 @@ import { GeneralFormFieldId } from './steps/general-information/constants';
+import { NetworkMapFieldId } from './steps/network-map/constants';
 import NetworkMapStep from './steps/network-map/NetworkMapStep';
 import OtherSettingsStep from './steps/other-settings/OtherSettingsStep';
+import ReviewStep from './steps/review/ReviewStep';
+import { StorageMapFieldId } from './steps/storage-map/constants';
 import StorageMapStep from './steps/storage-map/StorageMapStep';
+import { VmFormFieldId } from './steps/virtual-machines/constants';
 import VirtualMachinesStep from './steps/virtual-machines/VirtualMachinesStep';
 import { firstStep, planStepNames, planStepOrder, PlanWizardStepId } from './constants';
 import CreatePlanWizardFooter from './CreatePlanWizardFooter';
 import { useCreatePlanForm, useDefaultFormValues } from './hooks';
-import { hasWarmMigrationProviderType } from './utils';
+import { getCreatedPlanPath, handlePlanSubmission, hasWarmMigrationProviderType } from './utils';
 
 import './CreatePlanWizard.style.scss';
 
 const CreatePlanWizard: FC = () => {
   const { t } = useForkliftTranslation();
+  const history = useHistory();
+  const [currentStep, setCurrentStep] = useState<WizardStepType>(firstStep);
+  const [createPlanError, setCreatePlanError] = useState<Error>();
+  const [isCreating, setIsCreating] = useState(false);
+
   const defaultValues = useDefaultFormValues();
   const form = useCreatePlanForm({
     defaultValues,
     mode: 'onChange',
   });
-  const [currentStep, setCurrentStep] = useState<WizardStepType>(firstStep);
-  const { control, formState, watch } = form;
-  const formValues = watch();
-  const sourceProvider = useWatch({ control, name: GeneralFormFieldId.SourceProvider });
+  const hasCreatePlanError = Boolean(createPlanError?.message);
+
+  const { control, formState } = form;
+  const [planName, planProject, sourceProvider, targetProvider, vms, networkMap, storageMap] =
+    useWatch({
+      control,
+      name: [
+        GeneralFormFieldId.PlanName,
+        GeneralFormFieldId.PlanProject,
+        GeneralFormFieldId.SourceProvider,
+        GeneralFormFieldId.TargetProvider,
+        VmFormFieldId.Vms,
+        NetworkMapFieldId.NetworkMap,
+        StorageMapFieldId.StorageMap,
+      ],
+    });
+
+  const onSubmit = async () => {
+    setCreatePlanError(undefined);
+    setIsCreating(true);
+
+    try {
+      await handlePlanSubmission({
+        networkMap,
+        planName,
+        planProject,
+        sourceProvider,
+        storageMap,
+        targetProvider,
+        vms: Object.values(vms),
+      });
 
-  // TODO, Normalize wizard data object and submit
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  const onSubmit = () => {};
+      // Navigate to the created plan
+      history.push(getCreatedPlanPath(planName, planProject));
+    } catch (error) {
+      if (error instanceof Error) {
+        setCreatePlanError(error);
+      } else {
+        setCreatePlanError(new Error('An unknown error occurred'));
+      }
```

Comment:
const err = error as Error;
setCreatePlanError(error);

then u dont need the else, its safe as error is always unknown.